### PR TITLE
Introduced `runInSandbox` and added warnings when multiple instances of mobx are in a project.

### PR DESCRIPTION
**Pull request by mweststrate**
See #1082

### CODE REVIEW

File: src/core/globalstate.ts [hunk 1]
```diff
@@ -88,7 +88,34 @@ export class MobXGlobals {
...
+{
+	const global = getGlobal();
+	if (!global.__mobxInstanceCount) {
+		global.__mobxInstanceCount = 1;
+	} else {
+		global.__mobxInstanceCount++;
+		setTimeout(() => {
+			if (!shareGlobalStateCalled && !warnedAboutMultipleInstances && !_runInSandbox) {
+				warnedAboutMultipleInstances = true;
+				console.warn("[mobx] Warning: there are multiple mobx active instances active. This might lead to unexpected results. See https://github.com/mobxjs/mobx/issues/1082 for details.")
```
Comment (andykog) [inline src/core/globalstate.ts]: > active instances active 

typo :-)

**Comment by andykog** [comment]
Looks good

File: src/core/globalstate.ts [hunk 2]
```diff
@@ -88,7 +88,34 @@ export class MobXGlobals {
...
+			}
+		})
+	}
+}
+
+export function runInSandbox() {
```
Comment (urugator) [inline src/core/globalstate.ts]: Maybe reconsider the name, there is no concept of actual sandbox and there is `runInAction`, so someone may (I would) expect something more like this: 
```javascript
runInSandbox(() => {
  // runs isolated
} /*, sandbox/global/namespace/whatever*/)
```
`supressMultipleInstancesWarning` perhaps?

File: src/core/globalstate.ts [hunk 3]
```diff
@@ -88,7 +88,34 @@ export class MobXGlobals {
 
 export let globalState: MobXGlobals = new MobXGlobals();
 
+let shareGlobalStateCalled = false;
+let warnedAboutMultipleInstances = false;
+let _runInSandbox = false;
```
Comment (urugator) [inline src/core/globalstate.ts]: For the sake of consistency, could `_runInSandbox` be named `runInSandboxCalled` just like `shareGlobalStateCalled`...?

File: src/core/globalstate.ts (hunk 2 above)
Comment (mweststrate) [inline src/core/globalstate.ts]: Renamed to it to **isolateGlobalState**, which is a clearer counterpart to **shareGlobalState**

File: src/core/globalstate.ts (hunk 3 above)
Comment (mweststrate) [inline src/core/globalstate.ts]: Renamed to `runInIsolationCalled`

File: src/core/globalstate.ts (hunk 1 above)
Comment (mweststrate) [inline src/core/globalstate.ts]: Fixed :)

File: src/core/globalstate.ts (hunk 3 above)
Comment (urugator) [inline src/core/globalstate.ts]: I don't want to nitpick, but was it meant to be `isolateGlobalStateCalled`?

**Pull request merged**